superset/assets/javascripts/locales.jsx

```diff
@@ -0,0 +1,198 @@
+import Jed from 'jed';
+import React from 'react';
+import { sprintf } from 'sprintf-js';
+import { getTranslations } from './translations';
+import { getLanguage } from './explore/stores/getLanguage';
+
+let LOCALE_DEBUG = false;
+
+if (sessionStorage && sessionStorage.getItem('localeDebug') === '1') {
+  LOCALE_DEBUG = true;
+}
+
+export function setLocaleDebug(value) {
+  sessionStorage.setItem('localeDebug', value ? '1' : '0');
+}
+
+let i18n = null;
+
+export function setLocale(locale) {
+  const translations = getTranslations(locale);
+  i18n = new Jed({
+    domain: 'superset',
+    locale_data: {
+      superset: translations,
+    },
+  });
+}
+
+setLocale(getLanguage());
+
+function formatForReact(formatString, args) {
```

Comment:
This method is used to parse the incoming parameters
such as 'title={t('Preview for %s', query.tableName)}'